ip_reader.c

```diff
@@ -0,0 +1,200 @@
+#include <stdio.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+#include <ctype.h>
+#include "ip_reader.h"
+
+#define OCTET_1ST(ip) ((ip) >> 24)
+#define OCTET_2ND(ip) OCTET_1ST((ip) << 8)
+#define OCTET_3RD(ip) OCTET_1ST((ip) << 16)
+#define OCTET_4TH(ip) OCTET_1ST((ip) << 24)
+#define DECIMAL_FORMAT_STRING "%u.%u.%u.%u"
+#define DECIMAL_FORMAT_VALUES(ip) OCTET_1ST(ip),OCTET_2ND(ip),OCTET_3RD(ip),OCTET_4TH(ip)
```

Comment:
Still not what I wanted. My suggestion:
IPV4_OCTET_1 ...
IPV4_DECIMAL_FORMAT_STRING
IPV4_DECIMAL_FORMAT_VALUES
The whole point is to clearly say these are macros for IPv4.
Also using "st", "nd", etc. suffixes can be painful for programmers, don't do it.